CROSS-CUTTING-CONTRIBUTING.md

```diff
@@ -0,0 +1,75 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Creating a development environment
+
+When cross cutting changes are needed, it is best to work in a personal branch, but would need to replicate this to each submodule impacted.
+
+```
+$ git checkout -b CASSANDRA-<number>
+# Change the URL to point to your fork
+$ git submodule set-url accord "git@github.com:$USER/cassandra-accord.git"
+# Change the branch to your development branch
+$ (branch="$(git rev-parse --abbrev-ref HEAD)"; cd accord && git checkout -b "$branch")
+```
+
+To better automate this a utility script exists, so can do
+
+```
+$ dev-support/development-switch.sh --jira CASSANDRA-<number>
```

Comment:
I know my GH user is different than my `$USER`. Might be worth just mentioning `--github-user` here.

Reply:
I didn't as I don't think it's the avg case.  The script now asks for your GH user, so `--github-user` now only exists for cases where you want to override, such as if I want to work on Blakes project I do `bdeggleston` (though https will have issues...)

Reply:
https://stackoverflow.com/questions/40841882/automatically-access-git-submodules-via-ssh-or-https

Not sure if this simplifies things, but if we say the module must reside in the same source server as the c* branch referencing it we could use relative URLs and perhaps solve the user and https problem…